### PR TITLE
RD-6212 Export props interfaces together with their corresponding components

### PR DESCRIPTION
**Pull request by kubama**
## Description
Export props interfaces together with their corresponding components. These are then automatically re-exported from the top-level `index.ts` making them available for import directly from the package root, without the need for specyfying file path.

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [X] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [X] I added proper labels to this PR.

## Tests
Tested by installing locally built package in stage.

 

## Documentation
N/A

### CODE REVIEW

File: src/components/popups/PopupMenu/PopupMenu.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import React, { useState } from 'react';
 import { isEmpty } from 'lodash';
 import type { IconProps, PopupProps, SemanticICONS } from 'semantic-ui-react';
 import { Icon } from 'semantic-ui-react';
-import Popup from '../Popup';
+import { Popup } from 'components';
 import './PopupMenu.css';
 
 export interface PopupMenuProps extends Omit<IconProps, 'name'> {
```
Comment (qooban) [inline src/components/popups/PopupMenu/PopupMenu.tsx:9]: General - Docs:

I'm wondering maybe worth to add a simple bullet in "General guidelines for adding new components" in "Development" section in `/README.md` file that it is required / strongly recommended to export props type?

**Review by qooban — DISMISSED** [review]

File: test/Pagination.test.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import { mount } from 'enzyme';
 
-import Pagination from '../src/components/data/common/Pagination';
-import PaginationInfo from '../src/components/data/common/Pagination/PaginationInfo';
-import type { PaginationProps } from '../src/components/data/common/Pagination/Pagination';
+import Pagination from 'components/data/common/Pagination';
+import PaginationInfo from 'components/data/common/Pagination/PaginationInfo';
+import type { PaginationProps } from 'components/data/common/Pagination/Pagination';
```
[the remark this answers is not in the document]
Comment (kubama) [reply test/Pagination.test.tsx:6]: The `Pagination` component is not exported from the package (only used internally), so this shortcut wouldn't work in this case.

File: src/components/popups/PopupMenu/PopupMenu.tsx (hunk 1 above)
Comment (kubama) [reply src/components/popups/PopupMenu/PopupMenu.tsx:9]: Done

File: test/Pagination.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (qooban) [reply test/Pagination.test.tsx:6]: Oh, right, got it.

**Review by qooban — APPROVED** [review]

File: src/components/buttons/CancelButton/CancelButton.tsx [hunk 3]
```diff
@@ -1,18 +1,22 @@
 import React from 'react';
-import { Button } from 'semantic-ui-react';
 import type { ButtonProps } from 'semantic-ui-react';
+import { Button } from 'semantic-ui-react';
+
+export type CancleButtonProps = ButtonProps;
```
Comment (PatrycjaKubiczek) [inline src/components/buttons/CancelButton/CancelButton.tsx:5]: typo `CancleButtonProps`
```suggestion
export type CancelButtonProps = ButtonProps;
```

File: src/components/buttons/CancelButton/CancelButton.tsx [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
  * so all properties of that component (eg. onClick, disabled, ...) can be used here.
  */
-export default function CancelButton({
+export function CancelButton({
     content = 'Cancel',
     icon = 'remove',
     className = 'cancel',
     ...buttonProps
-}: ButtonProps) {
+}: CancleButtonProps) {
```
Comment (PatrycjaKubiczek) [inline src/components/buttons/CancelButton/CancelButton.tsx:18]: typo CancleButtonProps
```suggestion
}: CancelButtonProps) {
```

File: src/components/data/DataTable/DataTable.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import type { ReactNode } from 'react';
 import React, { Component } from 'react';
-import { isEmpty, debounce } from 'lodash';
+import { debounce, isEmpty } from 'lodash';
```
Comment (PatrycjaKubiczek) [inline src/components/data/DataTable/DataTable.tsx:3]: just out of curiosity: are you using some kind of extension to order imports alphabetically or do you do it manually?

**Review by PatrycjaKubiczek — CHANGES_REQUESTED** [review]

File: src/components/data/DataTable/DataTable.tsx (hunk 5 above)
Comment (kubama) [reply src/components/data/DataTable/DataTable.tsx:3]: WebStorm has this feature built-in

**Pull request merged**